dask_deltatable/write.py

```diff
@@ -202,26 +203,21 @@ def to_deltalake(
         max_partitions=max_partitions,
         meta=(None, object),
     )
-    final_name = "delta-commit"
-    dsk = {
-        (final_name, 0): (
-            _commit,
-            table,
-            written.__dask_keys__(),
-            table_uri,
-            schema,
-            mode,
-            partition_by,
-            name,
-            description,
-            configuration,
-            storage_options,
-            partition_filters,
-            custom_metadata,
-        )
-    }
-    graph = HighLevelGraph.from_collections(final_name, dsk, dependencies=(written,))  # type: ignore
-    result = Scalar(graph, final_name, "")
+    result = dask.delayed(_commit, name="deltatable-commit")(
```

Comment:
I'm not sure if this is doing what it's supposed to. I'll add a couple tests

Reply:
it's not calling optimise, so it's indeed not working properly

PR to fix: https://github.com/dask/dask/pull/11231